### DISCUSSION TITLE
Model gets repositioned in upward direction after some time.

### DISCUSSION THREAD

**Discussion by prerna-mohan** [google/model-viewer#4425] [Q&A] [answered]
### Description

This was observed in webxr.
The model gets placed initially, but after sometime or some camera movement it will reposition itself upwards in Y axis.
Ideally, once the object is placed, it should be locked there, shouldn't reposition like this.

Go to https://modelviewer.dev/examples/annotations/index.html
1. For the "Astronaut" model and enter the webxr.
2. Observe that after some phone camera movements like back and forth, it repositions upwards incorrectly(above the surface).

#### Live Demo

https://modelviewer.dev/examples/annotations/index.html

### Version

- model-viewer: vX.X.X

### Browser Affected

- [ ] Chrome, version: xx.x.xxxx.xx
- [ ] Edge
- [ ] Firefox
- [ ] IE

### OS

- [x] Android

### AR

- [x] WebXR

**Comment by prerna-mohan** [comment]
@elalish @cdata Can you please help here.

**Comment by elalish** [comment]
This is intended behavior - it can't place correctly until the floor is found by ARCore, which may never happen if it can't find the floor. We could block on this like Quick Look does, but that makes the experience even more broken (never seeing the model sometimes). I would recommend adding HTML instructions for your users, since you can scope your CSS to whether the floor has been found yet or not. See examples here: https://modelviewer.dev/examples/augmentedreality/index.html#webXR

**Reply by prerna-mohan** [reply]
Okay, but even if the floor is correctly found, then also it re-positions itself upwards(which is like few inches above the ground). I noticed that the event 'object-placed' isn't triggered untill it repositions like this. So does this mean it should necessarily re-position like this? If yes, then why it has to be above the ground so much?

**Reply by elalish** [reply, marked as answer]
I'd need to see a repro video to guess, but first off - ARCore estimates the floor height and it's not that accurate - a few inches may well be within the margin of error. It tends to improve over time. 

Sometimes something about the asset itself applies a transform that makes it difficult for us to calculate the right bounding box, and that can cause offsets; that's generally related to animations.
